File: Nodejs/tophub.js
const axios = require('axios');

class Tophub {
	constructor(apikey = '') {
		this.baseurl = 'https://api.tophubdata.com';
		this.apikey = apikey;
	}

	async nodes(p = 1) {
		return this.call('/nodes', { p });
	}

	async node(hashid) {
		return this.call(`/nodes/${hashid}`, {});
	}

	async nodeHistorys(hashid, date) {
		return this.call(`/nodes/${hashid}/historys`, { date });
	}

	async search(q, p = 1, hashid = '') {
		return this.call('/search', { q, p, hashid });
	}

	async call(endpoint, params = {}) {
		const url = `${this.baseurl}${endpoint}`;
		const headers = { 'Authorization': this.apikey };

		try {
			const response = await axios.get(url, { params, headers });
			return response.data;
		} catch (error) {
			console.error(`Request failed with status code ${error.response.status}`);
			return null;
		}
	}
}

// Example usage
const apikey = '';
const tophub = new Tophub(apikey);

(async () => {
	const resultNodes = await tophub.nodes(1);
	const resultNode = await tophub.node('mproPpoq6O');
	const resultNodeHistorys = await tophub.nodeHistorys('mproPpoq6O', '2023-01-01');
	const resultSearch = await tophub.search('苹果', 1, 'mproPpoq6O');
})();

File: Python/tophub.py
import requests

class Tophub:

	def __init__(self, apikey=''):
		self.baseurl = 'https://api.tophubdata.com'
		self.apikey = apikey

	def nodes(self, p=1):
		return self.call("/nodes", {"p": p})

	def node(self, hashid):
		return self.call(f"/nodes/{hashid}", {})

	def node_historys(self, hashid, date):
		return self.call(f"/nodes/{hashid}/historys", {"date": date})

	def search(self, q, p=1, hashid=''):
		return self.call("/search", {"q": q, "p": p, "hashid": hashid})

	def call(self, endpoint, params={}):
		url = f"{self.baseurl}{endpoint}"
		if params:
			url += "?" + "&".join([f"{key}={value}" for key, value in params.items()])

		headers = {'Authorization': self.apikey}
		response = requests.get(url, headers=headers)

		if response.status_code == 200:
			return response.json()
		else:
			print(f"Request failed with status code {response.status_code}")
			return None

# Example usage
apikey = ''
tophub = Tophub(apikey)
result_nodes = tophub.nodes(1)
result_node = tophub.node('mproPpoq6O')
result_node_historys = tophub.node_historys('mproPpoq6O', '2023-01-01')
result_search = tophub.search('苹果', 1, 'mproPpoq6O')
